Userland/Libraries/LibWeb/CSS/Parser/Parser.cpp

```diff
@@ -1688,6 +1691,103 @@ JS::GCPtr<CSSSupportsRule> Parser::convert_to_supports_rule(AtRule const& rule)
     return CSSSupportsRule::create(m_context.realm(), supports.release_nonnull(), rule_list);
 }
+JS::GCPtr<CSSPropertyRule> Parser::convert_to_property_rule(AtRule const& rule)
+{
+    // https://drafts.css-houdini.org/css-properties-values-api-1/#at-ruledef-property
+    // @property <custom-property-name> {
+    // <declaration-list>
+    // }
+
+    if (rule.prelude.is_empty()) {
+        dbgln_if(CSS_PARSER_DEBUG, "Failed to parse @property rule: Empty prelude.");
+        return {};
+    }
+
+    auto prelude_stream = TokenStream { rule.prelude };
+    prelude_stream.discard_whitespace();
+    auto const& token = prelude_stream.consume_a_token();
+    if (!token.is_token()) {
+        dbgln_if(CSS_PARSER_DEBUG, "CSSParser: @property has invalid prelude, prelude = {}; discarding.", rule.prelude);
+        return {};
+    }
+
+    auto name_token = token.token();
+    prelude_stream.discard_whitespace();
+
+    if (prelude_stream.has_next_token()) {
+        dbgln_if(CSS_PARSER_DEBUG, "CSSParser: @property has invalid prelude, prelude = {}; discarding.", rule.prelude);
+        return {};
+    }
+
+    if (!name_token.is(Token::Type::Ident)) {
+        dbgln_if(CSS_PARSER_DEBUG, "CSSParser: @property name is invalid: {}; discarding.", name_token.to_debug_string());
+        return {};
+    }
+
+    if (!name_token.ident().starts_with_bytes("--"sv)) {
+        dbgln_if(CSS_PARSER_DEBUG, "CSSParser: @property name doesn't start with '--': {}; discarding.", name_token.ident());
+        return {};
+    }
+
+    auto const& name = name_token.ident();
+
+    Optional<FlyString> syntax_maybe;
+    Optional<bool> inherits_maybe;
+    Optional<String> initial_value_maybe;
+
+    rule.for_each_as_declaration_list([&](auto& declaration) {
+        if (declaration.name.equals_ignoring_ascii_case("syntax"sv)) {
+            TokenStream token_stream { declaration.value };
+            token_stream.discard_whitespace();
+
+            auto const& syntax_token = token_stream.consume_a_token();
+            if (syntax_token.is(Token::Type::String)) {
+                token_stream.discard_whitespace();
+                if (token_stream.has_next_token()) {
+                    dbgln_if(CSS_PARSER_DEBUG, "CSSParser: Unexpected trailing tokens in syntax");
+                } else {
+                    syntax_maybe = syntax_token.token().string();
+                }
+            } else {
+                dbgln_if(CSS_PARSER_DEBUG, "CSSParser: Unexpected value for @property \"syntax\": {}; discarding.", declaration.to_string());
+            }
+            return;
+        }
+        if (declaration.name.equals_ignoring_ascii_case("inherits"sv)) {
+            TokenStream token_stream { declaration.value };
+            token_stream.discard_whitespace();
+
+            auto const& inherits_token = token_stream.consume_a_token();
+            // Feels hacky using Ident here, but this is what the tokenizer gives us.
```

Comment:
Nit-pick: We don't need this comment. (Any unquoted text is an `<ident>`, it's just how the grammar works.)